fuzzers/iso8601_fuzzer.c

```diff
@@ -6,40 +6,38 @@
  * This source code is licensed under the GNU Lesser General Public License
  * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
  */
+
 #include <stdint.h>
 #include <stdlib.h>
 #include <stdio.h>
 
 #include <crm/common/util.h>
-#include <crm/common/iso8601.h>
-#include <crm/common/iso8601_internal.h>
-
+#include <crm/common/internal.h>
 
 int
 LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
 {
-  char *ns;
-  char *result;
-  time_t epoch;
-  pcmk__time_hr_t *now;
+    char *ns = NULL;
+    char *result = NULL;
+    time_t epoch = 0;
+    pcmk__time_hr_t *now = NULL;
 
-  // Ensure we have enough data.
-  if (size < 10) {
-    return 0;
-  }
-  ns = malloc(size+1);
-  memcpy(ns, data, size);
-  ns[size] = '\0';
+    // Ensure we have enough data.
+    if (size < 10) {
+        return 0;
+    }
+    ns = pcmk__assert_alloc(1, size + 1);
+    memcpy(ns, data, size);
+    ns[size] = '\0';
 
-  crm_time_parse_period(ns);
-  pcmk__time_hr_new(ns);
+    crm_time_parse_period(ns);
+    pcmk__time_hr_new(ns);
 
-  epoch = 0;
-  now = NULL;
-  now = pcmk__time_hr_now(&epoch);
-  result = pcmk__time_format_hr(ns, now);
-  free(result);
+    epoch = 0;
+    now = pcmk__time_hr_now(&epoch);
+    result = pcmk__time_format_hr(ns, now);
+    free(result);
```

Comment:
I don't think it matters much. A different `now` would change only the `struct tm tm` that is only passed to `strftime()`, so changing it would mostly cover `strftime()` rather than `pcmk__time_format_hr()`. It would indirectly cover the response of `pcmk__time_format_hr()` to different `strftime()` outputs, but that doesn't seem worth much attention.

Reply:
Was more like contemplating over how to handle such cases using an apparently simple example.
Turned out not to be that simple of an example as we're missing functions that would turn a hr-formatted-string or epoch + us into pcmk__time_hr_t.

For the case we want to address something similar later my thoughts on how we could do that:
- partition the input using sections as random epoche and random us
- instead of partitioning the input we could use a hash-function - how was the interface of those coming with glib - to create epoche and us from the format-string
- epoche  -> crm_time_t -> use internal struct pcmk__time_us to fill useconds afterwards